src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1641,6 +1650,43 @@ internal override void CreateAsync()
 
             // Create reader thread and send first PSRP message.
             StartReaderThread(_stdOutReader);
+
+            // Monitor SSH process until connection is established.
+            _sshProcessTimer = new Timer(
+                callback: (_) =>
+                {
+                    if (_connectionEstablished)
+                    {
+                        // Monitor no longer needed.
+                        _sshProcessTimer?.Change(Timeout.Infinite, Timeout.Infinite);
+                        return;
+                    }
+
+                    bool sshTerminated = false;
+                    try
+                    {
+                        // Detect if SSH client process terminates prematurely.
+                        using (var sshProcess = System.Diagnostics.Process.GetProcessById(_sshProcessId))
+                        {
+                            sshTerminated = sshProcess == null || sshProcess.Handle == IntPtr.Zero || sshProcess.HasExited;
```

Comment:
does this also work?
```suggestion
                            sshTerminated = sshProcess.HasExited ?? true;
```

Reply:
I found out through trial and error that you have to check the Handle property as well.  Your suggestion could throw and I don't like relying on try/catch for a simple check so decided to do this.  There is no point checking sshProcess for null twice so I include that check once.